eng/pipelines/libraries/build-job.yml

```diff
@@ -162,6 +152,17 @@ jobs:
               tarCompression: $(tarCompression)
               artifactName: $(librariesBuildArtifactName)
               displayName: Build Assets
+          
+          - ${{ if containsValue(parameters.testBuildPlatforms, parameters.platform) }}:
```

Comment:
Why do you still need the librariesBuildArtifact ? I thought the only place you were using it was on the test libraries build right? Now that we are doing it on the same machine we can probably remove it since we will just zip this up and send it to helix already.

Reply:
ooohh we are not sending things to helix here yet right? Any reason why not doing that here as well to avoid one extra set of legs?

Reply:
No, coreclr also needs them. Coreclr uses libraries product build to build the runtime tests. 

Also, to run the tests we need the testhost which is zipped up in `run-tests` because we override the runtime when we want to run against a checked or mono runtime.

Reply:
> ooohh we are not sending things to helix here yet right? Any reason why not doing that here as well to avoid one extra set of legs?

Because we just build the tests for `Windows_x64`, `Linux_x64` and `OSX_x64` and then we run tests for all platforms + we also run tests against Mono Desktop and Coreclr checked, which we don't have a libraries product build against those targets. Instead what we do, is download the libraries product build, the test assets, the runtime, and merge them together and then run. So that actually gives us the ability to fan out more test runs, with less libraries product build and less libraries test builds.